sdk/cosmosdb/cosmos/test/internal/session.spec.ts

```diff
@@ -127,6 +127,44 @@ describe.skip("Session Token", function (this: Suite) {
     assert.ok(resource);
     assert.strictEqual(statusCode, 200);
   });
+
+describe("Integrated Cache Staleness", function (this: Suite) {
+  beforeEach(async function () {
+    await removeAllDatabases();
+
+  it.only("Should check if the max integrated cache staleness header is set and the value is correct.", async function () {
+    const dbId = addEntropy("maxIntegratedCacheTestDB");
+    const containerId = addEntropy("maxIntegratedCacheTestContainer");
+
+    const client = new CosmosClient({
+      endpoint,
+      key: masterKey,
+      consistencyLevel: ConsistencyLevel.Eventual,
+      plugins: [
+        {
+          on: "request",
+          plugin: async (context, next) => {
+            if (context.headers["x-ms-dedicatedgateway-max-age"]) {
+              assert.strictEqual(context.headers["x-ms-dedicatedgateway-max-age"].valueOf(), 2001);
+            }
+            console.log(context);
+            const response = await next(context);
+            return response;
+          },
+        },
+      ],
+    });
+
+    const { database } = await client.databases.createIfNotExists({
+      id: dbId,
+    });
+    const { container } = await database.containers.createIfNotExists({
+      id: containerId,
+    });
+
+    await container.read({ maxIntegratedCacheStaleness: 2001 });
```

Comment:
Can you please explain why integratedCacheStaleness matter for container read? 
Also can you please expand on what API's do the test needs to cover?

Reply:
Moved to feed options. This will only allow maxIntegratedCacheStaleness option on the item read.